lib/commands/web.js

```diff
@@ -9,71 +9,109 @@ const IPHONE_EXTRA_WEB_COORD_SCROLL_OFFSET = -15;
 const IPHONE_EXTRA_WEB_COORD_NON_SCROLL_OFFSET = 10;
 const IPHONE_WEB_COORD_OFFSET = -10;
 const IPHONE_WEB_COORD_SMART_APP_BANNER_OFFSET = 84;
+const IPHONE_X_EXTRA_WEB_COORD_SCROLL_OFFSET = -90;
+const IPHONE_X_EXTRA_WEB_COORD_NON_SCROLL_OFFSET = -10;
+const IPHONE_X_WEB_COORD_OFFSET = 40;
 const IPAD_EXTRA_WEB_COORD_SCROLL_OFFSET = -10;
 const IPAD_EXTRA_WEB_COORD_NON_SCROLL_OFFSET = 0;
 const IPAD_WEB_COORD_OFFSET = 10;
 const IPAD_WEB_COORD_SMART_APP_BANNER_OFFSET = 95;
 
+const IPHONE_X_WIDTH = 375;
+const IPHONE_X_HEIGHT = 812;
+
 let extensions = {};
 
 Object.assign(extensions, iosCommands.web);
 
 const getSafariIsIphone = _.memoize(async function getSafariIsIphone (sessionId, driver) {
   // sessionId parameter is for memoizing per session
-  let isIphone = true;
   try {
-    const useragent = await driver.execute('return navigator.userAgent');
-    isIphone = useragent.toLowerCase().includes('iphone');
+    const userAgent = await driver.execute('return navigator.userAgent');
+    return userAgent.toLowerCase().includes('iphone');
   } catch (err) {
     log.warn(`Unable to find device type from useragent. Assuming iPhone`);
     log.debug(`Error: ${err.message}`);
   }
-  return isIphone;
+  return true;
+});
+
+const getSafariIsIphoneX = _.memoize(async function getSafariIsIphone (sessionId, driver) {
```

Comment:
we could just make this a helper/extension and then memoize it on the constructor instead of passing sessionId around.